.github/workflows/ci.yaml

```diff
@@ -5,48 +5,50 @@ on:
     branches:
       - master
 env:
-  PYTHON_VERSION: 3.8
+  MIN_PYTHON_VERSION: 3.8
   PY_COLORS: 1
+  IS_MAIN_PYTHON: ${{ matrix.python-version == env.MIN_PYTHON_VERSION && matrix.platform == 'ubuntu-latest' }}
 
 jobs:
   test:
     name: Run tests
     strategy:
+      fail-fast: false
       matrix:
         platform: [ubuntu-latest, windows-latest]
         python-version: ["3.8", "3.9"]
     runs-on: ${{ matrix.platform }}
     steps:
       - uses: actions/checkout@v4
-      - name: Install Python tools
-        uses: BrandonLWhite/pipx-install-action@v0.1.1
-      - uses: actions/setup-python@v5
+      - name: Install Python
+        uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
-          cache: poetry
-
-      - name: Install PyGobject dependencies
-        if: matrix.platform == 'ubuntu-latest'
+          cache: 'poetry'
```

Comment:
This unfortunately does not work, since `poetry` needs to be installed before `actions/setup-python@v5` is pulled in. That's why there was `pipx-install-action` which sets up both `poetry` and `poe`.